diff --git a/glib/gconvert.c b/glib/gconvert.c
index 7078679..8dc85d2 100644
--- a/glib/gconvert.c
+++ b/glib/gconvert.c
@@ -21,18 +21,12 @@
 #include "config.h"
 #include "glibconfig.h"
 
-#ifndef G_OS_WIN32
 #include <iconv.h>
-#endif
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 
-#ifdef G_OS_WIN32
-#include "win_iconv.c"
-#endif
-
 #ifdef G_PLATFORM_WIN32
 #define STRICT
 #include <windows.h>
diff --git a/meson.build b/meson.build
index 03168cf..ea9bdee 100644
--- a/meson.build
+++ b/meson.build
@@ -1681,7 +1681,7 @@ glibconfig_conf.set10('G_HAVE_GROWING_STACK', growing_stack)
 # GNU implementation that ships with MinGW.
 
 # On Windows, just always use the built-in implementation
-if host_system == 'windows'
+if host_system == 'IGNOREME windows'
   libiconv = []
   glib_conf.set('USE_LIBICONV_NATIVE', true)
 else
diff --git a/glib/meson.build b/glib/meson.build
index 6c6786f..5e96a6d 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -280,12 +280,12 @@ libglib_dep = declare_dependency(
   link_with : libglib,
   # thread_dep doesn't get pulled in from libglib atm,
   # see https://github.com/mesonbuild/meson/issues/1426
-  dependencies : [thread_dep, libintl],
+  dependencies : [thread_dep, libintl] + libiconv,
   # We sadly need to export configinc here because everyone includes <glib/*.h>
   include_directories : [configinc, glibinc])
 
 pkg.generate(libglib,
-  libraries : [libintl],
+  libraries : [libintl] + libiconv,
   libraries_private : [osx_ldflags, win32_ldflags],
   subdirs : ['glib-2.0'],
   extra_cflags : ['-I${libdir}/glib-2.0/include'] + win32_cflags,
diff --git a/gio/win32/meson.build b/gio/win32/meson.build
index c1841cb..dc84cca 100644
--- a/gio/win32/meson.build
+++ b/gio/win32/meson.build
@@ -10,6 +10,6 @@ giowin32_sources = [
 giowin32_lib = static_library('giowin32',
   sources : [giowin32_sources],
   include_directories : [configinc, glibinc, gioinc, gmoduleinc],
-  dependencies : [libintl, gioenumtypes_dep],
+  dependencies : [libintl, gioenumtypes_dep] + libiconv,
   pic : true,
   c_args : [ '-DG_DISABLE_DEPRECATED' ] + gio_c_args)
