From 6e81923d2fca1ca070519ff5fc60172539eede59 Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwelk@xhochy.com>
Date: Thu, 11 Apr 2024 05:54:18 +0000
Subject: [PATCH] Remove broken private statement from comment

---
 glib/gspawn-win32.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/glib/gspawn-win32.c b/glib/gspawn-win32.c
index f14911b..3050aef 100644
--- a/glib/gspawn-win32.c
+++ b/glib/gspawn-win32.c
@@ -165,8 +165,7 @@ safe_wspawnvpe (int _Mode,
 
 #else
 
-/**< private >
- * ensure_cmd_environment:
+/* ensure_cmd_environment:
  *
  * Workaround for an issue in the universal C Runtime library (UCRT). This adds
  * a custom environment variable to this process's environment block that looks
