From 63571e235ef34d2bbb8bb15e95741fc240b04837 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwelk@xhochy.com>
Date: Sat, 6 Jul 2024 08:31:00 +0200
Subject: [PATCH] Skip Linux specific test on osx

---
 girepository/tests/repository.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/girepository/tests/repository.c b/girepository/tests/repository.c
index c79b0df..b74fbfa 100644
--- a/girepository/tests/repository.c
+++ b/girepository/tests/repository.c
@@ -837,7 +837,7 @@ test_repository_find_by_gtype (RepositoryFixture *fx,
     GType platform_specific_type;
     const char *expected_name, *expected_namespace;
 
-#if defined(G_OS_UNIX)
+#if defined(G_OS_UNIX) && !defined(__APPLE__)
     platform_specific_type = G_TYPE_DESKTOP_APP_INFO;
     expected_name = "DesktopAppInfo";
     expected_namespace = "GioUnix";
