From fea950b6bf46dbc70582fb56f889931586466bf8 Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Mon, 3 Sep 2018 22:29:49 -0400
Subject: [PATCH 2/3] Fix missing include paths in glib/gnulib.

---
 glib/gnulib/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/glib/gnulib/meson.build b/glib/gnulib/meson.build
index 6746631..44227c6 100644
--- a/glib/gnulib/meson.build
+++ b/glib/gnulib/meson.build
@@ -1,4 +1,4 @@
 gnulib_lib = static_library('gnulib', 'asnprintf.c', 'printf.c', 'printf-args.c', 'printf-parse.c', 'vasnprintf.c',
-  include_directories : configinc,
+  include_directories : [configinc, glibinc],
   pic : true,
   c_args : [ '-DLIBDIR="@0@"'.format(get_option('libdir')), '-DGLIB_COMPILATION', '-DG_LOG_DOMAIN="GLib"' ] + glib_hidden_visibility_args)
-- 
2.17.1

