diff --git a/glib/meson.build b/glib/meson.build
index 8350ea2..6c6786f 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -267,6 +267,7 @@ libglib = library('glib-2.0',
   soversion : soversion,
   darwin_versions : darwin_versions,
   install : true,
+  install_rpath : glib_libdir,
   # intl.lib is not compatible with SAFESEH
   link_args : [noseh_link_args, glib_link_flags, win32_ldflags],
   include_directories : configinc,
@@ -327,6 +328,7 @@ if host_system == 'windows'
 else
   gtester = executable('gtester', 'gtester.c',
     install : true,
+    install_rpath : glib_libdir,
     include_directories : configinc,
     dependencies : [libglib_dep])
 endif
diff --git a/gio/meson.build b/gio/meson.build
index d7f4f3f..5aa27b8 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -427,6 +427,7 @@ if host_system != 'windows'
 
     executable('gio-launch-desktop', 'gio-launch-desktop.c',
       install : true,
+      install_rpath : glib_libdir,
       c_args : gio_c_args,
       # intl.lib is not compatible with SAFESEH
       link_args : noseh_link_args)
@@ -803,6 +804,7 @@ libgio = library('gio-2.0',
   soversion : soversion,
   darwin_versions : darwin_versions,
   install : true,
+  install_rpath : glib_libdir,
   include_directories : [configinc, gioinc],
   #  '$(gio_win32_res_ldflag)',
   dependencies : [libz_dep, libdl_dep, libmount_dep, libglib_dep,
@@ -923,6 +925,7 @@ gio_tool_sources = [
 
 executable('gio', gio_tool_sources,
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -930,12 +933,14 @@ executable('gio', gio_tool_sources,
 
 executable('gresource', 'gresource-tool.c',
   install : true,
+  install_rpath : glib_libdir,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
   dependencies : [libelf, libgio_dep, libgobject_dep, libgmodule_dep, libglib_dep])
 
 gio_querymodules = executable('gio-querymodules', 'gio-querymodules.c', 'giomodule-priv.c',
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -944,6 +949,7 @@ gio_querymodules = executable('gio-querymodules', 'gio-querymodules.c', 'giomodu
 glib_compile_schemas = executable('glib-compile-schemas',
   [gconstructor_as_data_h, 'gvdb/gvdb-builder.c', 'glib-compile-schemas.c'],
   install : true,
+  install_rpath : glib_libdir,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
   dependencies : [libgio_dep, libgobject_dep, libgmodule_dep, libglib_dep])
@@ -951,6 +957,7 @@ glib_compile_schemas = executable('glib-compile-schemas',
 glib_compile_resources = executable('glib-compile-resources',
   [gconstructor_as_data_h, 'gvdb/gvdb-builder.c', 'glib-compile-resources.c'],
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -958,6 +965,7 @@ glib_compile_resources = executable('glib-compile-resources',
 
 executable('gsettings', 'gsettings-tool.c',
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -970,6 +978,7 @@ install_data(['gschema.loc', 'gschema.its'],
 
 executable('gdbus', 'gdbus-tool.c',
   install : true,
+  install_rpath : glib_libdir,
   c_args : gio_c_args,
   # intl.lib is not compatible with SAFESEH
   link_args : noseh_link_args,
@@ -978,6 +987,7 @@ executable('gdbus', 'gdbus-tool.c',
 if host_system != 'windows' and not glib_have_cocoa
   executable('gapplication', 'gapplication-tool.c',
     install : true,
+    install_rpath : glib_libdir,
     c_args : gio_c_args,
     # intl.lib is not compatible with SAFESEH
     link_args : noseh_link_args,
diff --git a/gobject/meson.build b/gobject/meson.build
index 159f33e..455e4c6 100644
--- a/gobject/meson.build
+++ b/gobject/meson.build
@@ -117,6 +117,7 @@ libgobject = library('gobject-2.0',
   soversion : soversion,
   darwin_versions : darwin_versions,
   install : true,
+  install_rpath : glib_libdir,
   include_directories : [configinc],
   dependencies : [libffi_dep, libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GLib-GObject"', '-DGOBJECT_COMPILATION'] + glib_hidden_visibility_args,
@@ -138,6 +139,7 @@ libgobject_dep = declare_dependency(link_with : libgobject,
 
 executable('gobject-query', 'gobject-query.c',
   install : true,
+  install_rpath : glib_libdir,
   dependencies : [libglib_dep, libgobject_dep])
 
 install_data('gobject_gdb.py', install_dir : join_paths(glib_pkgdatadir, 'gdb'))
diff --git a/gmodule/meson.build b/gmodule/meson.build
index e4a6a1c..14c60c7 100644
--- a/gmodule/meson.build
+++ b/gmodule/meson.build
@@ -101,6 +101,7 @@ libgmodule = library('gmodule-2.0',
   soversion : soversion,
   darwin_versions : darwin_versions,
   install : true,
+  install_rpath : glib_libdir,
   include_directories : [configinc, gmoduleinc],
   dependencies : [libdl_dep, libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GModule"', '-DG_DISABLE_DEPRECATED'] + glib_hidden_visibility_args,
diff --git a/gthread/meson.build b/gthread/meson.build
index d4fca40..043a259 100644
--- a/gthread/meson.build
+++ b/gthread/meson.build
@@ -18,6 +18,7 @@ libgthread = library('gthread-2.0',
   soversion : soversion,
   darwin_versions : darwin_versions,
   install : true,
+  install_rpath : glib_libdir,
   dependencies : [libglib_dep],
   c_args : ['-DG_LOG_DOMAIN="GThread"' ] + glib_hidden_visibility_args,
   link_args : glib_link_flags,
