diff --git a/gio/tests/resources.c b/gio/tests/resources.c
index 7ba5896..0b72278 100644
--- a/gio/tests/resources.c
+++ b/gio/tests/resources.c
@@ -640,7 +640,7 @@ test_resource_manual2 (void)
 static void
 test_resource_binary_linked (void)
 {
-  #ifndef __linux__
+  #ifndef __linux_and_objcopy_has_add_symbols__
   g_test_skip ("--external-data test only works on Linux");
   return;
   #else /* if __linux__ */
