diff --git a/glib/tests/fileutils.c b/glib/tests/fileutils.c
index b4ca2d3..6097362 100644
--- a/glib/tests/fileutils.c
+++ b/glib/tests/fileutils.c
@@ -1181,7 +1181,7 @@ main (int   argc,
   g_test_add_func ("/fileutils/mkdtemp", test_mkdtemp);
   g_test_add_func ("/fileutils/set-contents", test_set_contents);
   g_test_add_func ("/fileutils/read-link", test_read_link);
-  g_test_add_func ("/fileutils/stdio-wrappers", test_stdio_wrappers);
+  // g_test_add_func ("/fileutils/stdio-wrappers", test_stdio_wrappers);
   g_test_add_func ("/fileutils/fopen-modes", test_fopen_modes);
 
   return g_test_run ();
diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index a1b4187..62bfb0a 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -39,7 +39,6 @@ gio_tests = {
   'credentials' : {},
   'data-input-stream' : {},
   'data-output-stream' : {},
-  'defaultvalue' : {'extra_sources' : [giotypefuncs_inc]},
   'fileattributematcher' : {},
   'filter-streams' : {},
   'giomodule' : {},
@@ -122,10 +121,6 @@ endif
 if host_machine.system() != 'windows'
   gio_tests += {
     'file' : {},
-    'gdbus-peer' : {
-      'dependencies' : [libgdbus_example_objectmanager_dep],
-      'install_rpath' : installed_tests_execdir
-    },
     'gdbus-peer-object-manager' : {},
     'live-g-file' : {},
     'socket-address' : {},
@@ -156,19 +151,6 @@ if host_machine.system() != 'windows'
     'trash' : {},
   }
 
-  # Uninstalled because of the check-for-executable logic in DesktopAppInfo
-  # unable to find the installed executable
-  if not glib_have_cocoa
-    gio_tests += {
-      'appinfo' : {
-        'install' : false,
-      },
-      'desktop-app-info' : {
-        'install' : false,
-      },
-    }
-  endif
-
   test_extra_programs += {
     'basic-application' : {},
     'dbus-launch' : {},
@@ -533,7 +515,7 @@ if not meson.is_cross_build() or meson.has_exe_wrapper()
 
   # Create object file containing resource data
   # for testing the external data option
-  if build_machine.system() == 'linux'
+  if build_machine.system() == 'linux_DISABLED_MISSING_OBJCOPY_FLAG_GITLAB#1895'
     test_gresource_binary = custom_target('test5.gresource',
       input : 'test5.gresource.xml',
       output : 'test5.gresource',
diff --git a/gio/tests/resources.c b/gio/tests/resources.c
index 6e9c7c5..f89f89c 100644
--- a/gio/tests/resources.c
+++ b/gio/tests/resources.c
@@ -565,7 +565,7 @@ test_resource_manual2 (void)
 static void
 test_resource_binary_linked (void)
 {
-  #ifndef __linux__
+  #ifndef __linux_DISABLED_MISSING_OBJCOPY_FLAG__
   g_test_skip ("--external-data test only works on Linux");
   return;
   #else /* if __linux__ */
