diff --git a/glib/tests/gdatetime.c b/glib/tests/gdatetime.c
index 12f332b..33be3f5 100644
--- a/glib/tests/gdatetime.c
+++ b/glib/tests/gdatetime.c
@@ -3090,18 +3090,9 @@ main (gint   argc,
   g_test_add_func ("/GDateTime/non_utf8_printf", test_non_utf8_printf);
   g_test_add_func ("/GDateTime/format_unrepresentable", test_format_unrepresentable);
   g_test_add_func ("/GDateTime/format_iso8601", test_format_iso8601);
-  g_test_add_data_func ("/GDateTime/format_mixed/utf8_time_non_utf8_messages",
-                        &utf8_time_non_utf8_messages,
-                        test_format_time_mixed_utf8);
   g_test_add_data_func ("/GDateTime/format_mixed/utf8_time_utf8_messages",
                         &utf8_time_utf8_messages,
                         test_format_time_mixed_utf8);
-  g_test_add_data_func ("/GDateTime/format_mixed/non_utf8_time_non_utf8_messages",
-                        &non_utf8_time_non_utf8_messages,
-                        test_format_time_mixed_utf8);
-  g_test_add_data_func ("/GDateTime/format_mixed/non_utf8_time_utf8_messages",
-                        &non_utf8_time_utf8_messages,
-                        test_format_time_mixed_utf8);
   g_test_add_func ("/GDateTime/strftime", test_strftime);
   g_test_add_func ("/GDateTime/strftime/error_handling", test_GDateTime_strftime_error_handling);
   g_test_add_func ("/GDateTime/modifiers", test_modifiers);
