diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 81ff551..055cade 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -216,19 +216,6 @@ if host_machine.system() != 'windows'
     }
   endif
 
-  # Uninstalled because of the check-for-executable logic in DesktopAppInfo
-  # unable to find the installed executable
-  if not glib_have_cocoa
-    gio_tests += {
-      'appinfo' : {
-        'install' : false,
-      },
-      'desktop-app-info' : {
-        'install' : false,
-      },
-    }
-  endif
-
   test_extra_programs += {
     'basic-application' : {},
     'dbus-launch' : {},
@@ -414,19 +401,6 @@ if host_machine.system() != 'windows'
     },
   }
 
-  # Generate test.mo from de.po using msgfmt
-  msgfmt = find_program('msgfmt', required : false)
-  if msgfmt.found()
-    subdir('de/LC_MESSAGES')
-    gio_tests += {
-      'gsettings' : {
-        'extra_sources' : [test_mo],
-        'c_args' : ['-DSRCDIR="@0@"'.format(meson.current_source_dir()),
-                    '-DTEST_LOCALE_PATH="@0@"'.format(test_mo_dir)],
-        'install' : false,
-      },
-    }
-  endif
 endif # unix
 
 #  Test programs buildable on Windows only
