From 8ccdd8fd7e6a97969655cd3eecce1587ce3cea4f Mon Sep 17 00:00:00 2001
From: Uwe Korn <uwelk@xhochy.com>
Date: Tue, 9 Sep 2025 06:22:02 +0000
Subject: [PATCH 6/6] gio/tests/socket-listener: Depend on libdl

---
 gio/tests/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index c5f3f9c..8715870 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -130,7 +130,9 @@ gio_tests = {
     'can_fail' : host_system in ['darwin', 'gnu'],
     'suite': ['flaky'],
   },
-  'socket-listener' : {},
+  'socket-listener' : {
+    'dependencies': libdl_dep,
+  },
   'socket-service' : {},
   'srvtarget' : {},
   'task' : {},
